Source/SwiftLintBuiltInRules/Rules/Idiomatic/NoEmptyBlockRule.swift

```diff
@@ -102,13 +103,36 @@ private extension NoEmptyBlockRule {
             // No need to show a warning since Empty Block of `guard` is compile error.
             guard node.parent?.kind != .guardStmt else { return }
 
+            if let codeBlockType = node.codeBlockType, configuration.enabledBlockTypes.contains(codeBlockType) {
+                validate(node: node)
+            }
+        }
+
+        func validate(node: CodeBlockSyntax) {
             guard node.statements.isEmpty,
                   !node.leftBrace.trailingTrivia.containsComments,
                   !node.rightBrace.leadingTrivia.containsComments else {
                 return
             }
-
             violations.append(node.leftBrace.positionAfterSkippingLeadingTrivia)
         }
     }
 }
+
+private extension CodeBlockSyntax {
+    var codeBlockType: NoEmptyBlockConfiguration.CodeBlockType? {
+        guard let kind = self.parent?.kind else { return nil }
+        switch kind {
+        case .accessorDecl:
+            return .accessorBodies
+        case .functionDecl:
+            return .functionBodies
+        case .initializerDecl, .deinitializerDecl:
+            return .initializerBodies
+        case .forStmt, .doStmt, .whileStmt, .repeatStmt, .ifExpr, .guardStmt, .catchClause, .deferStmt:
+            return .statementBlocks
```

Comment:
```suggestion
        case .forStmt, .doStmt, .whileStmt, .repeatStmt, .ifExpr, .catchClause, .deferStmt:
            return .statementBlocks
        case .guardStmt:
            return nil
```

That way, you can skip the check in line 104.